### ISSUE TITLE
Using yum module with state: latest & update_only: yes shows contradicting output

### ISSUE THREAD

**Issue by SWADESNA** [ansible/ansible#72061]
##### SUMMARY

When "update_only" parameter is set to yes with yum module the results shows "All packages providing jre1.8 are up to date", while it does updates the package to the latest version.
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

yum
##### ANSIBLE VERSION

```paste below
ansible 2.9.11
```

##### CONFIGURATION

```paste below
N/A
```

##### OS / ENVIRONMENT

RHEL 7 & RHEL6 on target machine

##### STEPS TO REPRODUCE

1. Install lower version of jre on target host (eg. jre1.8.x86_64 0:1.8.0_251-fcs)

2. Below sample playbook task:

```yaml

    - name: Install the latest pkg
      yum:
        name: jre
        state: latest
        update_only: yes
```

##### EXPECTED RESULTS

yum should update the jre package to the latest available version ( jre1.8.x86_64 0:1.8.0_261-fcs  ) & show the same in result

##### ACTUAL RESULTS

yum updates the package to the latest while the result  shows    "All packages providing jre1.8 are up to date", in the start.

Attaching output of the same:

```
{
    "_ansible_no_log": false,
    "changed": true,
    "results": [
        "All packages providing jre1.8 are up to date",
        "Loaded plugins: product-id, search-disabled-repos, subscription-manager\n\nThis system is not registered with an entitlement server. You can use subscription-manager to register.\n\nResolving Dependencies\n--> Running transaction check\n---> Package jre1.8.x86_64 0:1.8.0_251-fcs will be updated\n---> Package jre1.8.x86_64 0:1.8.0_261-fcs will be an update\n--> Finished Dependency Resolution\n--> Finding unneeded leftover dependencies\nFound and removing 0 unneeded dependencies\n\nDependencies Resolved\n\n================================================================================\n Package     Arch        Version              Repository                   Size\n================================================================================\nUpdating:\n jre1.8      x86_64      1.8.0_261-fcs        xxx-rhel-7-java_oracle       69 M\n\nTransaction Summary\n================================================================================\nUpgrade  1 Package\n\nTotal download size: 69 M\nDownloading packages:\nDelta RPMs disabled because /usr/bin/applydeltarpm not installed.\nRunning transaction check\nRunning transaction test\nTransaction test succeeded\nRunning transaction\n  Updating   : jre1.8-1.8.0_261-fcs.x86_64                                  1/2 \nUnpacking JAR files...\n\tplugin.jar...\n\tjavaws.jar...\n\tdeploy.jar...\n\trt.jar...\n\tjsse.jar...\n\tcharsets.jar...\n\tlocaledata.jar...\n  Cleanup    : jre1.8-1.8.0_251-fcs.x86_64                                  2/2 \nwarning: file /usr/java/jre1.8.0_251-amd64/lib/javasettings: remove failed: No such file or directory\nwarning: file /usr/java/jre1.8.0_251-amd64/lib/install.jar: remove failed: No such file or directory\n  Verifying  : jre1.8-1.8.0_261-fcs.x86_64                                  1/2 \n  Verifying  : jre1.8-1.8.0_251-fcs.x86_64                                  2/2 \n\nUpdated:\n  jre1.8.x86_64 0:1.8.0_261-fcs                                                 \n\nComplete!\n"
    ],
    "attempts": 1,
    "rc": 0,
    "invocation": {
        "module_args": {
            "install_weak_deps": true,
            "autoremove": false,
            "lock_timeout": 30,
            "download_dir": null,
            "install_repoquery": true,
            "update_cache": true,
            "conf_file": null,
            "exclude": [
                ""
            ],
            "update_only": true,
            "installroot": "/",
            "allow_downgrade": false,
            "name": [
                "jre1.8"
            ],
            "download_only": false,
            "bugfix": false,
            "list": null,
            "disable_gpg_check": false,
            "disable_excludes": null,
            "use_backend": "auto",
            "validate_certs": true,
            "state": "latest",
            "disablerepo": [
                "*"
            ],
            "releasever": null,
            "disable_plugin": [],
            "enablerepo": [
                "rhel-7-server-rpms",
                "rhel-7-server-extras-rpms",
                "rhel-7-server-supplementary-rpms",
                "rhel-7-server-optional-rpms",
                "xxx-rhel-7-core",
                "xxx-rhel-7-java_oracle"
            ],
            "skip_broken": false,
            "security": false,
            "enable_plugin": []
        }
    },
    "changes": {
        "updated": [],
        "installed": []
    },
    "msg": ""
}
```

```paste below

```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/yum.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/yum.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by egarbi** [comment]
its worth to mention too that in certain circumstances when trying to update the package mentioned above without using `update_only: yes` the yum module simple failed to see a newer version of the package.
I thought first that it was a cache issue, but after cleaning up, the issue persists.
it was not until I added the `update_only` parameter that the issue was gone but anyway showing the contradictory results mentioned above.

PD: A RedHat's guy opened this issue for me, but I was who was impacted

**Comment by mkihr** [comment]
Same issue here with 2.9.12. Whatever package I try, state:latest always reports "All packages providing ..... are up to date." No matter if it's true or not.

**Comment by ursetto** [comment]
Same here, back on 2.9.1 — packages not upgrading to latest.

This is when installing on a system with the **subscription-manager plugin enabled, but not registered**. (You can see this is also the case in the original issue description.) To be clear, this is when you are installing a package from a repository not handled by subscription-manager.

It can be worked around by adding `disable_plugin: subscription-manager` to the `yum` module call, but this is not ideal. Alternatively you could disable the plugin on the system, but if you don't notice it's enabled, you will have no idea your packages are not being upgraded to the latest version. It just silently says `ok`.

My guess is the `This system is not registered with an entitlement server. You can use subscription-manager to register.` message is confusing the parser.

**Comment by egarbi** [comment]
Confirmed what @ursetto pointed out. Disabling subscription-manager cause will prevent this issue to happen.

**Comment by mkrizek** [comment]
I believe this has been fixed in https://github.com/ansible/ansible/pull/75452. If not feel free to open a new issue.
